# Iris Dataset Analysis with NumPy

This notebook demonstrates basic data analysis techniques using NumPy on the classic Iris dataset. We calculate statistical measures like mean, median, standard deviation, and percentiles to understand the distribution of sepal length.

---

## 🔹 Step 1: Load the Dataset

In [1]:
import numpy as np

# Load the dataset from UCI repository
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data'
iris = np.genfromtxt(url, delimiter=',', dtype='object')

# Preview the first few rows
print(iris[:5])

[[b'5.1' b'3.5' b'1.4' b'0.2' b'Iris-setosa']
 [b'4.9' b'3.0' b'1.4' b'0.2' b'Iris-setosa']
 [b'4.7' b'3.2' b'1.3' b'0.2' b'Iris-setosa']
 [b'4.6' b'3.1' b'1.5' b'0.2' b'Iris-setosa']
 [b'5.0' b'3.6' b'1.4' b'0.2' b'Iris-setosa']]


---

## 🔹 Step 2: Extract Sepal Length Column

We focus on the first column (sepal length) for statistical analysis.

In [2]:
sepallength = np.genfromtxt(url, delimiter=',', dtype='float', usecols=[0])
print("First 5 sepal length values:", sepallength[:5])

First 5 sepal length values: [5.1 4.9 4.7 4.6 5. ]


---

## 🔹 Step 3: Calculate Basic Statistics

In [3]:
mean = np.mean(sepallength)
median = np.median(sepallength)
std_dev = np.std(sepallength)

print("Mean:", mean)
print("Median:", median)
print("Standard Deviation:", std_dev)

Mean: 5.843333333333334
Median: 5.8
Standard Deviation: 0.8253012917851409


---

## 🔹 Step 4: Calculate Percentiles

We compute the 95th percentile to understand the upper range of sepal length.

In [4]:
percentile_95 = np.percentile(sepallength, 95)
print("95th Percentile of Sepal Length:", percentile_95)

95th Percentile of Sepal Length: 7.254999999999998


---

## ✅ Summary

- Mean Sepal Length: `5.84`
- Median Sepal Length: `5.8`
- Standard Deviation: `~0.83`
- 95th Percentile: `~7.25`

This analysis gives a quick overview of the distribution of sepal length in the Iris dataset. You can extend this by analyzing other features or visualizing the data using matplotlib or seaborn.